File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<Character> stack=new Stack<>();
        int ans=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                ans++;
                stack.push('(');
            }
            else if(s.charAt(i)==')'){
                if(stack.isEmpty() || stack.peek()!='('){
                    stack.push(')');
                    ans++;
                }
                else{
                    stack.pop();
                    ans--;
                }
            }
        }
        return ans;
    }
}

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.c
int minAddToMakeValid(char* s) {
    size_t n=strlen(s);
    int count=0;
    int ans=0;
    for(int i=0;i<n;i++){
        if(s[i]=='('){
            ans++;
        }
        else{
            ans--;
        }
        if(ans<0){
            count++;
            ans=0;
        }
    }
    return count+ans;
}